tools/SampleAnalyzer/Process/Counter/Counter.h

```diff
@@ -1,114 +1,128 @@
-  std::pair<MAint64,MAint64> nentries_;
-
-  /// sum of weights
-  /// first = positive weight ; second = negative weight
-  std::pair<MAfloat64,MAfloat64> sumweight_;
-
-  /// sum of squared weights
-  /// first = positive weight ; second = negative weight
-  std::pair<MAfloat64,MAfloat64> sumweight2_;
-
-
-  // -------------------------------------------------------------
-  //                       method members
-  // -------------------------------------------------------------
- public :
-
-  /// Constructor without argument 
-  Counter(const std::string& name = "unkwown")
-  { 
-    name_       = name;
-    nentries_   = std::make_pair(0,0); 
-    sumweight_  = std::make_pair(0.,0.);
-    sumweight2_ = std::make_pair(0.,0.);
-  }
-
-  /// Destructor
-  ~Counter()
-  { }
-
-  /// Reset
-  void Reset()
-  {
-    nentries_   = std::make_pair(0,0); 
-    sumweight_  = std::make_pair(0.,0.);
-    sumweight2_ = std::make_pair(0.,0.);
-  }
-
-  /// Increment the counter
-  void Increment(const MAfloat32& weight=1.)
-  {
-    if (weight>0)
-    {
-      nentries_.first++;
-      sumweight_.first+=weight;
-      sumweight2_.first+=weight*weight;
-    }
-    else if (weight<0)
+    class Counter
     {
-      nentries_.second++;
-      sumweight_.second+=weight;
-      sumweight2_.second+=weight*weight;
-    }
-  }
-
-};
+        friend class CounterManager;
+
+        // -------------------------------------------------------------
+        //                        data members
+        // -------------------------------------------------------------
+    public:
+        /// name of the analysis
+        std::string name_;
+
+        /// number of times the function Increment is called
+        /// first = positive weight ; second = negative weight
+        std::map<MAint32, ENTRIES> nentries_;
+
+        /// sum of weights
+        /// first = positive weight ; second = negative weight
+        std::map<MAint32, WEIGHTS> sumweights_;
+
+        /// sum of squared weights
+        /// first = positive weight ; second = negative weight
+        std::map<MAint32, WEIGHTS> sumweights2_;
+
+        // -------------------------------------------------------------
+        //                       method members
+        // -------------------------------------------------------------
+    public:
+        /// Constructor without argument
+        Counter(const std::string &name = "unkwown")
+        {
+            name_ = name;
+            Reset();
+        }
+
+        /// Destructor
+        ~Counter() {}
+
+        /// Reset
+        void Reset()
+        {
+            nentries_.clear();
+            sumweights_.clear();
+            sumweights2_.clear();
+        }
+
+        MAint32 size() { return nentries_.size(); }
+
+        void Initialise(const WeightCollection &multiweight)
+        {
+            Reset();
+            for (auto &weight : multiweight.GetWeights())
+            {
+
+                nentries_[weight.first] = ENTRIES();
```

Comment:
To be updated: `nentries_ = ENTRIES();`

Reply:
you are right entries does not need multiweight